typed/typed.go

```diff
@@ -128,40 +128,32 @@ func NewTypedData(types map[string]TypeDef, pType string, dom Domain) (td TypedD
 // Parameters:
 // - account: A pointer to a big.Int representing the account.
 // - msg: A TypedMessage object representing the message.
-// - sc: A StarkCurve object representing the curve.
 // Returns:
 // - hash: A pointer to a big.Int representing the calculated hash.
-// - err: An error object indicating any error that occurred during the calculation.
-func (td TypedData) GetMessageHash(account *big.Int, msg TypedMessage, sc curve.StarkCurve) (hash *big.Int, err error) {
+func (td TypedData) GetMessageHash(account *big.Int, msg TypedMessage) (hash *big.Int) {
 	elements := []*big.Int{utils.UTF8StrToBig("StarkNet Message")}
 
-	domEnc, err := td.GetTypedMessageHash("StarkNetDomain", td.Domain, sc)
-	if err != nil {
-		return hash, fmt.Errorf("could not hash domain: %w", err)
-	}
+	domEnc := td.GetTypedMessageHash("StarkNetDomain", td.Domain)
+
 	elements = append(elements, domEnc)
 	elements = append(elements, account)
 
-	msgEnc, err := td.GetTypedMessageHash(td.PrimaryType, msg, sc)
-	if err != nil {
-		return hash, fmt.Errorf("could not hash message: %w", err)
-	}
+	msgEnc := td.GetTypedMessageHash(td.PrimaryType, msg)
 
 	elements = append(elements, msgEnc)
-	hash, err = sc.ComputeHashOnElements(elements)
-	return hash, err
+	hash = curve.ComputeHashOnElements(elements)
+	return hash
```

Comment:
return curve.ComputeHashOnElements(elements)
```suggestion
	hash = curve.ComputeHashOnElements(elements)
	return hash
```